Match style of site overview in site context

Imran92 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/wp-calypso/issues/97960

## Proposed Changes

- Hide the header when Site Overview is opened in site-context
- Match the design overall with the domain-context view

https://github.com/user-attachments/assets/5d9be466-e260-46f3-8615-3f74c7b89680

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

- Update design for when Domain management revamp is opened in site-context

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- Enable the feature flag by `window.sessionStorage.setItem('flags', 'calypso/all-domain-management')`
- Purchase a blog domain and attach a site with it.
- Now go to calypso.localhost:3000/sites and open this site in flyout
- Scroll to bottom and click on the row to open the domain in site context
- Ensure that the site overview page looks the same as when it's not in site-context
- Make sure no design in breaking in domain-context

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Imran92 committed:
Hide header when needed

Imran92 committed:
Use site context param to hide header in flyout view

Imran92 committed:
Move style to be loaded in all views

phcp reviewed (changes requested):
For the style matching, I just noticed one thing missing, which is the border-radius on the right main container.

But I also found a navigation regression related to the site-context feature; if you go to the site dashboard > Upgrades > Domains and select a domain, it takes you out of the dashboard to the domains page with the site context rendered. Feel free not to include a fix on this PR, we can open a follow-up task if you prefer.

Imran92 committed:
Fix top border after removing header

Imran92 committed:
Fix wrong page context opening when flag enabled

Imran92 commented:
Thanks @phcp !

> For the style matching, I just noticed one thing missing, which is the border-radius on the right main container.

Good catch! I've fixed it here https://github.com/Automattic/wp-calypso/pull/98095/commits/888ef4a0bd6564194e01ccd1ae976c740d106ad9

> But I also found a navigation regression related to the site-context feature; if you go to the site dashboard > Upgrades > Domains and select a domain, it takes you out of the dashboard to the domains page with the site context rendered. Feel free not to include a fix on this PR, we can open a follow-up task if you prefer.

Thanks! Fixed it here https://github.com/Automattic/wp-calypso/pull/98095/commits/c8dd5fe87980b1aa4df3a2ef8d125293d8751341

Imran92 committed:
Fix rebase error

Imran92 commented:
Sorry for the force push, had conflicts after rebasing

phcp reviewed (approved):
Overall it looks good! There is only one minor thing, the container height overflows the screen size, so there is no spacing on the bottom. I'm leaving this approve, feel free to merge it and fix in a follow-up PR.
<img width="1010" alt="image" src="https://github.com/user-attachments/assets/c59a0dc5-4608-4aa1-97cb-f98c95ef287b" />

Imran92 committed:
Fix email upsell page styles in site context (#98152)

* Move style to be loaded in all views

* Fix email upsell page styles in site context

* Fix context menu not loading

* Fix unit test failing

Imran92 committed:
Fix scroll spacing issue

Imran92 commented:
> Overall it looks good! There is only one minor thing, the container height overflows the screen size, so there is no spacing on the bottom. I'm leaving this approve, feel free to merge it and fix in a follow-up PR.

I looked into it a bit, and it seemed like an issue stemming from the new breadcrumb. I added a quick fix below, but I'll create an issue to properly look into it and fix

https://github.com/Automattic/wp-calypso/pull/98095/commits/b26f8b068d1fdb7cf4868e5da0112293725d6da9

alshakero commented on client/my-sites/domains/domain-management/domain-overview-pane/style.scss:
```diff
@@ -148,3 +148,42 @@
+	}
+
+	.hosting-dashboard-item-view__content div {
+		flex-grow: 1;
```
Hi @Imran92! This prop is quite generic, and it's affecting all my divs in the domain side of things. Can you narrow it to your needs please?

Imran92 replied:
Hey @alshakero 👋 It should be alright now, we've removed that style
